src/Controller/AdminController.php

```diff
@@ -139,6 +156,74 @@ public function check(RequestDataBag $dataBag): JsonResponse
         }
     }
 
+    /**
+     * Send a capture request to the Adyen platform
+     *
+     * @Route(
+     *     "/api/adyen/capture",
+     *     name="api.adyen_payment_capture.post",
+     *     methods={"POST"}
+     * )
+     *
+     * @param Request $request
+     * @return JsonResponse
+     */
+    public function sendCaptureRequest(Request $request)
+    {
+        $context = Context::createDefaultContext();
+        $orderId = $request->request->get('orderId');
+        // If payload does not contain orderNumber
+        if (empty($orderId)) {
+            $message = 'Order Id was not provided in request';
+            $this->logger->error($message);
+            return new JsonResponse($message, 400);
+        }
+
+        /** @var OrderEntity $order */
+        $order = $this->orderRepository->getOrder($orderId, $context, ['currency']);
+
+        if (is_null($order)) {
+            $message = sprintf('Unable to find order %s', $orderId);
+            $this->logger->error($message);
+            return new JsonResponse($message, 400);
+        }
+
+        $currencyIso = $order->getCurrency()->getIsoCode();
+        $amountInMinorUnit = $this->currencyUtil->sanitize($order->getAmountTotal(), $currencyIso);
+
+        try {
+            $results = $this->captureService
+                ->doOpenInvoiceCapture($order->getOrderNumber(), $amountInMinorUnit, $context);
+        } catch (CaptureException $e) {
+            $this->logger->error($e->getMessage());
+
+            return new JsonResponse([
+                'success' => false,
+                'message' => 'adyen.error',
+            ]);
+        }
+
+        return new JsonResponse(['success' => true, 'response' => $results]);
```

Comment:
If an exception is thrown, is `$results` still defined?

Reply:
no it is not, but then the exception will be returned and this line will not be executed